source/beamline/CLS/CLSBeamlineStatus.cpp

```diff
@@ -40,6 +40,14 @@ bool CLSBeamlineStatus::isOff() const
 	}
 }
 
+AMControl * CLSBeamlineStatus::beamlineStatusPVControl()
+{
+	if (beamlineStatusPVControl_ )
+		return beamlineStatusPVControl_ ;
+	else
+		return this;
```

Comment:
the usage exists in ` void CLSBeamlineStatusView::refresh()' and the changes  are

{ 	
// Update the beam status editor.
 - if (editor_) 	
-     editor_->setControl(beamlineStatus_); 	
 + if (beamStatusEditor_) {
+     beamStatusEditor_->setControl(beamlineStatus_->beamlineStatusControl());
+ }
```
I think my changes fit the usage here.